src/osd/PrimaryLogPG.cc

```diff
@@ -424,6 +424,8 @@ void PrimaryLogPG::on_local_recover(
       derr << __func__ << " " << hoid << " had no clone_snaps" << dendl;
     }
   }
+  // temporary assert, see https://tracker.ceph.com/issues/50608
+  ceph_assert(recovery_info.version);
```

Comment:
Is this change intended for releasing with Tentacle? Or it's just about gathering more info from Teuthology?

Reply:
I was hoping we could backport it along with the first commit. Looking again, it would be probably safer to backport the first commit only, so I'll drop this additional assert.
If this issue was to be caught by Teuthology we would have enough information from the logs.

Reply:
@rzarzynski, updated. Do you think we can backport it after qa runs?